### ISSUE TITLE
The .dartServer cache leads to persistent failure of the analyzer

### ISSUE THREAD

**Issue by Hixie** [dart-lang/sdk#28466]
I just got into a situation where no matter what branch I was in, no matter what files I edited, no matter what I did, even `git clean -xffd` of my entire repo, `flutter analyze --watch --flutter-repo` gave me hundreds of bogus messages.

I ended up having to manually delete the `.dartServer` directory (which I only discovered because I ran the analyzer under `strace -f` to figure out what files it was opening) to clear up the mess.

**Comment by scheglov** [comment]
Yes, analysis server stores results in the file system, and if a file has the same import/export closure, then its analysis results are the same, and errors can be returned from the file system. So, changes to random files might invalidate and cause recomputation of errors in files that have the changed file in their import/export closure, but will not affect the errors for other files.

Do you know what caused the bogus errors initially? Can you give examples of bogus messages?

**Comment by Hixie** [comment]
There were literally hundreds but at least some were errors that no longer applied because I'd changed the `strong-mode` section of the `.analysis_options` file being used.

**Comment by scheglov** [comment]
Ah... That's interesting.

We mix in some flag of AnalysisOptions into the cache key, including the `strong-mode` flag.
But I'm not aware about the `strong-mode` section. 
Do you mean [this section](https://github.com/flutter/flutter/blob/master/.analysis_options#L20)?
If not, could you give an example?

It is still worrisome if there are other kinds of errors, that cannot be explained by changes in `.analysis_options`. It would help if we could get details about these other errors.

**Comment by Hixie** [comment]
See https://www.dartlang.org/guides/language/analysis-options#specifying-strong-mode

**Comment by Hixie** [comment]
It may be they were all because of that. That would certainly be a good thing. :-)

**Comment by scheglov** [comment]
OK
Thank you for the info about strong mode options.

I will start with this CL, because it should be done anyway.
[Including error processors into the signature.](https://codereview.chromium.org/2651513002)

**Comment by scheglov** [comment]
[And the change to the strong mode options.](https://codereview.chromium.org/2648153002)

**Comment by scheglov** [comment]
I hope that these two changes fix the problem.

**Comment by Hixie** [comment]
I just ran into this again. This change: https://github.com/flutter/flutter/pull/8853
...didn't get understood as a semantic change. I had to delete my `.dartServer` directory to get it to work.

**Comment by scheglov** [comment]
The full version of SDK?

**Comment by devoncarew** [comment]
It looks like it's `1.23.0-dev.9.0` (from https://github.com/flutter/flutter/blob/master/bin/internal/dart-sdk.version).

**Comment by devoncarew** [comment]
@Hixie, are you switching between branches that use different versions of the dart sdk for analysis? `flutter analyze --watch` doesn't restart itself if the dart-sdk.version files changes - I'm wondering if that's the issue.

**Comment by Hixie** [comment]
I definitely sometimes switch branches across versions, but I always restart the analyzer to see if that fixes it, if there are unexpected errors like this.

**Comment by eyalkutz** [comment]
I don't understand how to clear the dart analyzer cache. please help it keeps crashing

**Comment by bwilkerson** [comment]
As per #44566, you can simply delete the directory `~/.dartServer` (preferably while the analysis server isn't running).

**Comment by eyalkutz** [comment]
@bwilkerson excuse me for being a noob but what does "~ " mean in a file path?
(I don't know if this is relevant but I am on windows)

**Comment by bwilkerson** [comment]
On Unix based OS's, the `~` is an abbreviation for the users home directory.

**Comment by eyalkutz** [comment]
@bwilkerson  I cannot find a file or folder named `.dartServer` at my home directory. Can you tell me where it is on windows?

**Comment by bwilkerson** [comment]
I didn't realize there was a difference. On Windows it should be a subdirectory of the directory that's the value of `LOCALAPPDATA`.

**Comment by eyalkutz** [comment]
thank you @bwilkerson. unfortunately, that did not fix the problem as the analysis server keeps crushing regardless of this. I will post an issue with the logs

**Comment by Krysl** [comment]
I had a similar problem.
I mistakenly used uppercase in the filename, then I corrected it to lowercase, at which point the analyzer reported an error saying there was an error in the file with the uppercase filename. 
I deleted the `.dartServer` folder, but the error continued to appear until I deleted the subfolder in `%APPDATA%\Code\User\workspaceStorage` that corresponds to VSCode's project folder. 
(Hope this helps someone with the same problem.)
